src/coreclr/ToolBox/superpmi/superpmi/jitinstance.cpp

```diff
@@ -365,11 +368,16 @@ JitInstance::Result JitInstance::CompileMethod(MethodContext* MethodToCompile, i
             pParam->pThis->mc->cr->recAllocGCInfoCapture();
 
             pParam->pThis->mc->cr->recMessageLog("Successful Compile");
+
+            pParam->metrics->SuccessfulCompiles++;
+            pParam->metrics->NumCodeBytes += NCodeSizeBlock;
```

Comment:
@BruceForstall Yes, but if it passes in "base" then we write its size to `NumCodeBytes` and we don't know that it will fail in diff, so we ignore asm files as you say but this new metric does not look at asm files if I read the change correctly.

So we pass `--override-total-base-metric` as total number of bytes from compiled methods with base and `--override-total-diff-metric` as total number of bytes from compiled methods with diff.

Reply:
I see, we actually are adding the code size before we do the asmdiff. It seems like we should delay until after `InvokeNearDiffer` is successful (either shows no diffs or some diffs, but doesn't crash) before adding the code size. @jakobbotsch ?

Reply:
Sure, that might make sense. But if there are misses then posting the summary is misleading in any case because it does not show any diff in those methods, which is probably where the diffs will actually be. So at least this way it is visible to me that there are roughly 7 KB of code that SPMI misses now and that you should probably post PMI diffs instead.

I can change it but I don't see how we can make this case not be misleading to the reader of the summary.

Reply:
I guess it has to be wordy, like
```
Total metric of base: 7178021 bytes, 101 methods;
Total metric of diff: 7171065 bytes, 100 methods;
Total metric of delta: -6956 bytes differs (-0.10% of base), -1 method (-1% of base)
Total metric of delta in matching methods: +100 (+0.1 of base) on 100 methods.
```

Reply:
But the numbers are not useful, since what I would be interested in as a reader is the diffs in the failed methods. The idea behind this change was to get a better overview of how the change affects libraries crossgen/PMI as a whole and we can never get that when there's missing data. When I see the "Missing data encountered" I always use PMI for my diffs instead.

Reply:
We could sum the code bytes in the dasm files in jit-analyze and at least give a clear warning that they do not account for all diffs. But this only helps if someone does not notice that SPMI diffs are only partial and then posts the results anyway, not sure if this is what we're afraid of? 
Are there are cases where we know there won't be diffs in the functions with missing data encountered so that the SPMI diffs give an accurate view anyway? I can see keeping track of a "number diffed code bytes" metric as @BruceForstall suggested for this case, but in other cases it seems we should not use SPMI for diffs.